mscore/importmxmlnotepitch.cpp

```diff
@@ -0,0 +1,177 @@
+                        qDebug("invalid step '%s'", qPrintable(step));        // TODO
+                  }
+            else if (e.name() == "display-octave") {
+                  const auto oct = e.readElementText();
+                  bool ok;
+                  _displayOctave = oct.toInt(&ok);
+                  if (!ok || _displayOctave < 0 || _displayOctave > 9) {
+                        //logError(QString("invalid octave '%1'").arg(strOct));
+                        qDebug("invalid octave '%s'", qPrintable(oct));       // TODO
+                        _displayOctave = -1;
+                        }
+                  }
+            else
+                  e.skipCurrentElement();                   // TODO log
+            }
+      }
+
+//---------------------------------------------------------
+//   pitch
+//---------------------------------------------------------
+
+/**
+ Parse the /score-partwise/part/measure/note/pitch node.
+ */
+
+void mxmlNotePitch::pitch(QXmlStreamReader& e)
+      {
+      Q_ASSERT(e.isStartElement() && e.name() == "pitch");
+
+      // defaults
+      _step = -1;
+      _alter = 0;
+      _octave = -1;
+
+      while (e.readNextStartElement()) {
+            if (e.name() == "alter") {
+                  const auto alter = e.readElementText();
+                  bool ok;
+                  _alter = MxmlSupport::stringToInt(alter, &ok);       // fractions not supported by mscore
+                  if (!ok || _alter < -2 || _alter > 2) {
+                        _logger->logError(QString("invalid alter '%1'").arg(alter), &e);
+                        bool ok2;
+                        const auto altervalue = alter.toDouble(&ok2);
+                        if (ok2 && (qAbs(altervalue) < 2.0) && (_accType == AccidentalType::NONE)) {
+                              // try to see if a microtonal accidental is needed
+                              _accType = microtonalGuess(altervalue);
+                              }
+                        _alter = 0;
+                        }
+                  }
+            else if (e.name() == "octave") {
+                  const auto oct = e.readElementText();
+                  bool ok;
+                  _octave = oct.toInt(&ok);
+                  if (!ok || _octave < 0 || _octave > 9) {
+                        _logger->logError(QString("invalid octave '%1'").arg(oct), &e);
+                        _octave = -1;
+                        }
+                  }
+            else if (e.name() == "step") {
+                  const auto step = e.readElementText();
+                  const auto pos = QString("CDEFGAB").indexOf(step);
+                  if (step.size() == 1 && pos >=0 && pos < 7)
+                        _step = pos;
+                  else
+                        _logger->logError(QString("invalid step '%1'").arg(step), &e);
+                  }
+            else
+                  ;       // TODO skipLogCurrElem();
```

Comment:
Results in compiler warning:
warning: suggest braces around empty body in an 'else' statement [-Wempty-body]